packages/firestore-compat/src/api/database.ts

```diff
@@ -989,6 +989,10 @@ export class Query<T = PublicDocumentData>
     this._userDataWriter = new UserDataWriter(firestore);
   }
 
+  get _query() {
+    return this._delegate._query;
+  }
+
```

Comment:
pardon my ignorance. Why this is needed?

Reply:
As part of the code path for getting the count, it assumes that it has the _query object cause I think that is how the api has evolved? I am guessing this value used to be on the delegate pre v9 but I don't know for sure? This just exposes it so that the code here will not throw:

https://github.com/firebase/firebase-js-sdk/blob/ee871fc0b157e1a186c2895b4290d70c8d1b986f/packages/firestore/src/core/count_query_runner.ts#L39

Reply:
Ah! Got it.